torch/nn/functional.py

```diff
@@ -648,8 +648,11 @@ def max_pool1d_with_indices(
     # See: https://github.com/pytorch/pytorch/pull/62544#issuecomment-896195121
     # and https://github.com/pytorch/pytorch/issues/62545 for context
     if ceil_mode != return_indices:
-        warnings.warn("Note that order of the arguments: ceil_mode and return_indices will change"
-                      "to match the args list in nn.MaxPool1d in a future release.")
+        # Only raise the warning if it isn't raised already in a single run
+        if (max_pool1d_with_indices.warning_count == 0):
```

Comment:
There's `warnings.simplefilter('once', DeprecationWarning)` if you want your warning to be raised once. It's possible to wrap it in the context manager so that it doesn't affect any other warnings.

Reply:
Thanks, Ivan!

Reply:
Tried:

```python
warnings.simplefilter("once", category=DeprecationWarning)
    if ceil_mode != return_indices:
        warnings.warn("not equal", DeprecationWarning)
```

But it still spams in the tests, not sure why that would be!

Reply:
The previous approach was the correct one. `warnings` does not work well with this sort of warnings.

Reply:
The previous approach, unfortunately, didn't work well with JIT, I'll try again if there is a fix for that.